ramips: DTS: VoCore2

p-wassi opened the pull request:
Maintainer: @blogic 
Compile tested: ramips/mt7628/VoCore2
Run tested: ramips/mt7628/VoCore2

The VoCore2 features 128MB of DDR RAM, like seen in the first commit [1],
the manufacturer's website [2] and the relevant file in the manufacturer's source code [3]
Currently, only 64MB are active/enabled, so change 0x4000000 to 0x8000000 in the memory
section of the DTS. This makes all 128MB available.

Further, the only LED this board is equipped with, is connected between the GPIO and GND.
Schematic at [4]. Therefore flag this LED as ACTIVE_HIGH.

The device's serial console is configured for ttyS2, which also shares pins with PWM.
If the pinmux configuration is left like it is now, kernel 4.9 shows:
`of_serial: probe of 10000e00.uart2 failed with error -22`
The serial console doesn't work with kernel 4.9, which is quite a pain for a device like the VoCore.
This thing, however, is fine with kernel 4.4 !
With the proposed change in the pinmux, I get:
`10000e00.uart2: ttyS2 at MMIO 0x10000e00 (irq = 30, base_baud = 2500000) is a 16550A`
`console [ttyS2] enabled`
and the console is finally working as expected.

This PR is split in two commits:
* fix memory and LED: intended for trunk and 17.01
* fix serial console: just for trunk (kernel 4.9), as it is fine on 17.01 (kernel 4.4)

---

Further things to mention / discuss:
* sysupgrade does not work with trunk (after instantly finishing the sysupgrade command, the device just reboots without having changed anything)
* Do we really need to specify the pinmux configuration for UART pins in VOCORE2.dts. For me, it looks like this configuration is redundant, as it's already done in the mt7628an.dtsi include file.
* The only LED is triggered by netdev/wlan0 _and_ used as the status LED in diag.sh at the same time. Can this be a problem?
* WiFi is heavily unstable, but that's really another issue


[1]:
https://git.lede-project.org/?p=source.git;a=commit;h=3f31029b190417e12f4d19d9999cd52ec76d0f85

[2]:
http://vocore.io/v2.html

[3]:
http://vonger.cn/misc/vocore2/openwrt.tar.bz2

[4]:
http://vonger.cn/misc/vocore2/source.zip

mkresin commented:
Hey @p-wassi, it is fine for me to squash both commits into one. Would you please change the pinmux node to:

```
&pinctrl {
	state_default: pinctrl0 {
		gpio {
			ralink,group = "wled_an";
			ralink,function = "gpio";
		};
	};
};
```

It should be all that is required. The gpio group is Pin#44 == gpio 12 on the second gpio bank.

bobafetthotmail commented:
Wasn't it supposed to be 2 versions (a Lite with 64MB/8MB flash and a full with 128MB/16MB flash)? https://www.indiegogo.com/projects/vocore2-4-coin-sized-linux-computer-with-wifi#/

From that page it seems there are at least 475 Vocore Lite sold to people. (also on main site there is a mention of Vocore2 Lite having 64MB in the firmware list)

mkresin commented:
Good spot @bobafetthotmail. Since the dts has a partition map matching a 16MB flash chip, I'm quite sure it is really the VoCore2 (non lite) that is supported by this dts.

p-wassi commented:
Thanks for the comments, the commits are now squashed into one.

@mkresin : The pinmux node is now changed according to your post above, and tested on the VoCore. Everything's still working and fine :)
EDIT: Oh, wait.. there's been a change to the pinmux code. Will be back in a moment.

@bobafetthotmail : The only version sold via the website (now) is the 'full' version with 128MB of RAM.
There may have been a LITE version during the indiegogo campaign. This should then probably go into a distinct DTS.

p-wassi commented:
@mkresin : Ok, here we go. Updated pinmux configuration is successfully tested on VoCore2.

mkresin commented:
@ndoo Any comments?

pinney commented:
An old patch for the VoCore Lite :
can be found here https://patchwork.ozlabs.org/patch/759392/

mkresin commented:
Thanks! Merged to my [staging tree](https://git.lede-project.org/?p=lede/mkresin/staging.git;a=summary).
